frontend/src/components/common/Resource/ScaleButton.tsx

```diff
@@ -0,0 +1,222 @@
+  }
+
+  const applyFunc = React.useCallback(updateFunc, [item]);
+
+  function handleSave(numReplicas: number) {
+    const cancelUrl = location.pathname;
+    const itemName = item.metadata.name;
+
+    setOpenDialog(false);
+
+    // setOpenDialog(false);
+    dispatch(
+      clusterAction(() => applyFunc(numReplicas), {
+        startMessage: t('Scaling {{ itemName }}…', { itemName }),
+        cancelledMessage: t('Cancelled scaling {{ itemName }}.', { itemName }),
+        successMessage: t('Scaled {{ itemName }}.', { itemName }),
+        errorMessage: t('Failed to scale {{ itemName }}.', { itemName }),
+        cancelUrl,
+        errorUrl: cancelUrl,
+        ...options,
+      })
+    );
+  }
+
+  function handleClose() {
+    setOpenDialog(false);
+  }
+
+  React.useEffect(() => {
+    if (item) {
+      item
+        .getAuthorization('update')
+        .then((result: any) => {
+          if (result.status.allowed) {
+            setVisible(true);
+          }
+        })
+        .catch((err: Error) => {
+          console.error(`Error while getting authorization for edit button in ${item}:`, err);
+          setVisible(false);
+        });
+    }
+  }, [item]);
+
+  if (!visible || !['Deployment', 'StatefulSet', 'ReplicaSet'].includes(item.kind)) {
+    return null;
+  }
+
+  return (
+    <React.Fragment>
+      <Tooltip title={t('frequent|Scale') as string}>
+        <IconButton aria-label={t('frequent|scale')} onClick={() => setOpenDialog(true)}>
+          <Icon icon="mdi:overscan" />
+        </IconButton>
+      </Tooltip>
+      <ScaleDialog resource={item} open={openDialog} onClose={handleClose} onSave={handleSave} />
+    </React.Fragment>
+  );
+}
+
+interface ScaleDialogProps extends DialogProps {
+  resource: KubeObject;
+  onSave: (numReplicas: number) => void;
+  onClose: () => void;
+  errorMessage?: string;
+}
+
+const Input = styled(MuiInput)({
+  '& input[type=number]': {
+    MozAppearance: 'textfield',
+    textAlign: 'center',
+  },
+  '& input::-webkit-outer-spin-button, & input::-webkit-inner-spin-button': {
+    display: 'none',
+  },
+  width: '80px',
+});
+
+const useScaleDialogStyle = makeStyles(() => ({
+  dialogContent: {
+    paddingBottom: '30px', // Prevent the content from overflowing
+  },
+  replicasNumberInput: {
+    marginLeft: '6px',
+    marginRight: '6px',
+  },
+  replicasSwitcher: {
+    padding: '6px',
+    textAlign: 'left',
+  },
+  currentReplicasText: {
+    minWidth: '250px',
+  },
+}));
+
+function ScaleDialog(props: ScaleDialogProps) {
+  const { open, resource, onClose, onSave } = props;
+  const [numReplicas, setNumReplicas] = React.useState<number>(getNumReplicas());
+  const { t } = useTranslation(['frequent', 'resource']);
+  const classes = useScaleDialogStyle();
+  const theme = useTheme();
+  const desiredNumReplicasLabel = 'desired-number-replicas-label';
+  const numReplicasForWarning = 100;
+
+  function getNumReplicas() {
+    if (!resource?.spec) {
+      return -1;
```

Comment:
Why is -1 used here?

Reply:
Hmmm, it would indicate something weird happened (if we end up applying this to an unwanted resource).
I will change the label according to this value in the UI.